File: src/member/router.ts
import { Router } from 'express';

import {
  handleGetMembers,
  handleGetMember,
  handleCreateMember,
  handleUpdateMember,
  handleDeleteMember,
} from './controller';
import authMiddleware from '../auth/middleware';

const memberRouter = Router();

memberRouter.get('/', handleGetMembers);
memberRouter.get('/:id', handleGetMember);
memberRouter.post('/', authMiddleware, handleCreateMember);
memberRouter.put('/:id', authMiddleware, handleUpdateMember);
memberRouter.delete('/:id', authMiddleware, handleDeleteMember);

export default memberRouter;

File: members/router.go
package members

import (
	"github.com/AndreyGalchevski/strident-api/auth"
	"github.com/gorilla/mux"
)

func InitMembersRouter(r *mux.Router) {
	r.HandleFunc("/members", handleGetMembers).Methods("GET")

	r.HandleFunc("/members/{id}", auth.VerifyAuthorization(handleGetMemberByID)).Methods("GET")
	r.HandleFunc("/members", auth.VerifyAuthorization(handlePostMember)).Methods("POST")
	r.HandleFunc("/members/{id}", auth.VerifyAuthorization(handlePatchMember)).Methods("PATCH")
	r.HandleFunc("/members/{id}", auth.VerifyAuthorization(handleDeleteMember)).Methods("DELETE")
}
